XSConsoleData.py

```diff
@@ -965,10 +980,12 @@ def ManagementNetmask(self, inDefault = None):
 
         # FIXME: Address should come from API, but not available at present.  For DHCP this is just a guess at the gateway address
         for pif in self.derived.managementpifs([]):
-            if pif['ip_configuration_mode'].lower().startswith('static'):
+            ipv6 = pif['primary_address_type'].lower() == 'ipv6'
+            configuration_mode = pif['ipv6_configuration_mode'] if ipv6 else pif['ip_configuration_mode']
+            if configuration_mode.lower().startswith('static'):
                 # For static IP the API address is correct
-                retVal = pif['netmask']
-            elif pif['ip_configuration_mode'].lower().startswith('dhcp'):
+                retVal = pif['IPv6'][0].split('/')[1] if ipv6 else pif['netmask']
+            elif configuration_mode.lower().startswith('dhcp'):
                 # For DHCP,  find the gateway address by parsing the output from the 'route' command
```

Comment:
The comment is wrong. It is copied from the `ManagementGateway` function, but this code gets the `netmask` (for IPv4) or the `prefix length` (for IPv6) of the management interface:

https://docs.netgate.com/pfsense/en/latest/network/ipv6/subnets.html
> IPv6 doesn't have a subnet mask but instead calls it a Prefix Length, often shortened to “Prefix”.
https://www.ibm.com/docs/en/ts3500-tape-library?topic=formats-subnet-masks-ipv4-prefixes-ipv6
> Subnet masks (IPv4) and prefix lengths (IPv6) identify a range of IP addresses that are on the same network.

I think it would be good to not display "Netmask: 64" on the XSConsole status dislay but "Prefix length: 64" in the case of IPv6, but this is merely a UI nitpick.

The real thing about to comment here if the regular expression used below to read the IPv4 netmask from /sbin/ifconfig is indeed working to tread the IPv6. I doubt that:
```py
wlp9s0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.0.112  netmask 255.255.255.0  broadcast 192.168.0.255
        inet6 fe80::a1f1:9a23:6b57:11f4  prefixlen 64  scopeid 0x20<link>
        inet6 2001:4bb8:2d2:342c:360f:2c44:198e:d48a  prefixlen 64  scopeid 0x0<global>
```
Also, like for getting the gateway IP, here it must also be noted that XAPI supports the `autoconf` method for IPv6 address configuration and to read the prefix length from the interface like before. Likewise, the it should be added that `ifconfig` is obsolete and `ip` should be used for new code.

Reply:
Thanks, this is now addressed in #11